Fix ClassCastException from return value validation with proxy when adaptConstraintViolations=true

imvtsl opened the pull request:
Fix argument in call to applyReturnValueValidation() method in MethodValidationInterceptor.java. Method argument was passed instead of the return value of the method that was being validated.

Closes gh-33092

imvtsl committed:
Fix return value validation

Fix argument in call to applyReturnValueValidation()
method in MethodValidationInterceptor.java. Method
argument was passed instead of the return value of the
method that was being validated.

Closes gh-33092

pivotal-cla commented:
@imvtsl Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-framework&pullRequestId=33105)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-framework&pullRequestId=33105) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

pivotal-cla commented:
@imvtsl Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-framework&pullRequestId=33105)!

imvtsl commented:
@rstoyanchev
Please review.

rstoyanchev reviewed (changes requested):
It would be good to have a test. Would you like to add one, possibly in `MethodValidationProxyReactorTests` or `MethodValidationProxyTests`?

imvtsl commented:
Sure. I am working on adding test now.

imvtsl committed:
Fix return value validation

Fix argument in call to applyReturnValueValidation()
method in MethodValidationInterceptor.java. Method
argument was passed instead of the return value of the
method that was being validated.

Closes gh-33092

imvtsl commented:
@rstoyanchev 
I added tests to MethodValidationProxyTests.
In tests, I changed expected exception from ValidationException to ConstraintViolationException as ConstraintViolationException is more specific. I hope that's okay?

I read [CONTRIBUTING.md](https://github.com/spring-projects/spring-framework/blob/main/CONTRIBUTING.md#submit-a-pull-request) and I have a query. Should I sqaush these 2 commits into one or is it okay?

rstoyanchev commented:
I've squashed them for you and updated the tests a bit.
